core/pinger/pinger.go

```diff
@@ -73,13 +73,31 @@ func (pt *Pinger) loop() error {
 	}
 }
 
-// NoopPinger implements the pinger interface but just does nothing
-type NoopPinger struct{}
+// NoopPinger implements the pinger interface, does nothing but waits for heat
+// death of the universe.
+type NoopPinger struct {
```

Comment:
Now that all watchers have to behave like workers, you can't just return a thing that does nothing. It actually has to do something. This is the consequence of the new registry managing the lifecycle of the worker.